cpp/ql/lib/semmle/code/cpp/models/implementations/Allocation.qll

```diff
@@ -294,3 +386,99 @@ private class NewArrayAllocationExpr extends AllocationExpr, NewArrayExpr {
 
   override predicate requiresDealloc() { not exists(this.getPlacementPointer()) }
 }
+
+private module HeuristicAllocation {
+  /** A class that maps an `AllocationExpr` to an `HeuristicAllocationExpr`. */
+  private class HeuristicAllocationModeled extends HeuristicAllocationExpr instanceof AllocationExpr {
```

Comment:
Do we actually need this? I'd expect it to be handled by `HeuristicAllocationFunctionModeled`